projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -226,16 +242,39 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
-    public performTab(currentRowEl, rowIndex, visibleColumnIndex) {
-        if (!this.grid.rowList.find(row => row.index === rowIndex + 1) && this.grid.parent &&
-        this.grid.unpinnedColumns[this.grid.unpinnedColumns.length - 1].visibleIndex === visibleColumnIndex) {
+    public performTab(currentRowEl, rowIndex, visibleColumnIndex, isSummaryRow = false) {
+        const summaryRows = this.grid.summariesRowList.toArray();
+        const hasSummaries = summaryRows.length > 0;
+        const isLastDataRow = rowIndex === this.grid.verticalScrollContainer.igxForOf.length - 1;
+        const nextIsDataRow = this.grid.dataRowList.find(row => row.index === rowIndex + 1) ;
+        const isLastColumn =  this.grid.unpinnedColumns[this.grid.unpinnedColumns.length - 1].visibleIndex === visibleColumnIndex;
+        const isLastSummaryRow = hasSummaries && isSummaryRow;
+        if (!nextIsDataRow && !(isLastDataRow && hasSummaries) && isLastColumn && !isSummaryRow) {
+            // navigating in child, next is not summary
             this.navigateDown(currentRowEl, rowIndex, 0);
+        } else if (isLastSummaryRow && isLastColumn && this.grid.parent) {
+            // navigating in child summary, next is parent summary or next parent row
+            const parent = this.grid.parent;
+            const parentHasSummary = parent.summariesRowList.toArray().length > 0;
+            const parentRowIndex = parseInt(
+                this.getClosestElemByTag(currentRowEl, 'igx-child-grid-row').parentNode.getAttribute('data-rowindex'), 10);
+            const isLastRowInParent = parent.verticalScrollContainer.igxForOf.length - 1 === parentRowIndex;
+            if (isLastRowInParent && parentHasSummary) {
+                const parentSummary = parent.summariesRowList.toArray()[0].nativeElement;
+                parent.navigation.focusNextRow(parentSummary, 0, this.grid.rootGrid, true);
+            } else {
+                this.focusNext(0);
+            }
+        } else  if (isLastDataRow && hasSummaries && isLastColumn && this.grid.parent) {
+            // navigating in child rows, next is child grid's summary row
+           this.focusNextRow(summaryRows[0].nativeElement, 0, this.grid.parent, true);
         } else {
-            super.performTab(currentRowEl, rowIndex, visibleColumnIndex);
+            // navigating in normal cells
+            super.performTab(currentRowEl, rowIndex, visibleColumnIndex, isSummaryRow);
         }
     }
-    public performShiftTabKey(currentRowEl, rowIndex, visibleColumnIndex) {
-        if (visibleColumnIndex === 0 && rowIndex === 0 && this.grid.parent) {
+    public performShiftTabKey(currentRowEl, rowIndex, visibleColumnIndex, isSummary = false) {
+        if (visibleColumnIndex === 0 && rowIndex === 0 && this.grid.parent && !isSummary) {
             if (this.grid.allowFiltering) {
                 this.moveFocusToFilterCell();
             } else {
```

Comment:
In this scenario there should be also a check if the sibling child grid has summaries when pressing Shift + Tab to focus the last summary.